test/kubetest/k8s.go

```diff
@@ -718,6 +718,21 @@ func (k8s *K8s) DescribePod(pod *v1.Pod) {
 	}
 }
 
+//ShouldNotRestart checks that pod should not restart
+func (k8s *K8s) ShouldNotRestart(pod *v1.Pod, secondsForWait time.Duration) error {
+	containerStatuses := pod.Status.ContainerStatuses
+	for start := time.Now(); time.Since(start) < time.Second*secondsForWait; {
+		for _, container := range containerStatuses {
```

Comment:
Also, this loop will make a copy of `ContainerStatus` on each iteration. Let's use the indexing loop and pointers.